sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/ChangeFeedEncryptionProcessorBuilder.java

```diff
@@ -0,0 +1,135 @@
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to build a encryption supported {@link ChangeFeedProcessor} instance.
+ *
+ * {@codesnippet com.azure.cosmos.changeFeedProcessor.builder}
+ */
+public class ChangeFeedEncryptionProcessorBuilder {
+
+    private String hostName;
+    private CosmosEncryptionAsyncContainer feedContainer;
+    private CosmosAsyncContainer leaseContainer;
+    private ChangeFeedProcessorOptions changeFeedProcessorOptions;
+    private Consumer<List<JsonNode>> encryptionConsumer;
+
+    /**
+     * Helper class to build a encryption supported {@link ChangeFeedProcessor} instance.
+     *
+     */
+    public ChangeFeedEncryptionProcessorBuilder() {
+    }
+
+    /**
+     * Sets the host name.
+     *
+     * @param hostName the name to be used for the host. When using multiple hosts, each host must have a unique
+     * name.
+     * @return current Builder.
+     */
+    public ChangeFeedEncryptionProcessorBuilder hostName(String hostName) {
+        this.hostName = hostName;
+        return this;
+    }
+
+    /**
+     * Sets an existing {@link CosmosEncryptionAsyncContainer} to be used to read from the monitored container.
+     *
+     * @param feedContainer the instance of {@link CosmosEncryptionAsyncContainer} to be used.
+     * @return current Builder.
+     */
+    public ChangeFeedEncryptionProcessorBuilder feedContainer(CosmosEncryptionAsyncContainer feedContainer) {
+        this.feedContainer = feedContainer;
+
+        return this;
+    }
+
+    /**
+     * Sets an existing {@link CosmosAsyncContainer} to be used to read from the leases container.
+     *
+     * @param leaseContainer the instance of {@link CosmosAsyncContainer} to use.
+     * @return current Builder.
+     */
+    public ChangeFeedEncryptionProcessorBuilder leaseContainer(CosmosAsyncContainer leaseContainer) {
+        this.leaseContainer = leaseContainer;
+
+        return this;
+    }
+
+    /**
+     * Sets a consumer function which will be called to process changes.
+     *
+     * {@codesnippet com.azure.cosmos.changeFeedProcessor.handleChanges}
+     *
+     * @param consumer the {@link Consumer} to call for handling the feeds.
+     * @return current Builder.
+     */
+    public ChangeFeedEncryptionProcessorBuilder handleChanges(Consumer<List<JsonNode>> consumer) {
+        this.encryptionConsumer = jsonNodes -> {
+            List<byte[]> byteArrayList = jsonNodes.stream()
+                .map(node -> feedContainer.cosmosSerializerToStream(node))
+                .collect(Collectors.toList());
+            List<Mono<byte[]>> byteArrayMonoList =
+                jsonNodes.stream().map(jsonNode -> feedContainer.decryptResponse((ObjectNode) jsonNode)).collect(Collectors.toList());
+            Flux.concat(byteArrayMonoList).publishOn(Schedulers.boundedElastic()).map(
+                item -> feedContainer.getItemDeserializer().parseFrom(JsonNode.class, item)
+            ).collectList().doOnSuccess(consumer).block();
```

Comment:
We have to do blocking call as change feed processor using synchronous error handling to kill the worker. https://github.com/Azure/azure-sdk-for-java/blob/6fef0a30aeab08cb6a9c3db75a26bfdf2b6b16fc/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/DefaultObserver.java#L40

Reply:
thanks for explanation.

Later this consumer may be invoked on the io thread. Does that cause any problem (thread deadlock, etc)? or the fact that you are using publishOn takes care of that?

@milismsft is there any other workaround?

Reply:
@moderakh  yes for the same concern of block on IO thread i moved the publisher on bounded elastic where we can do the blocking call. This will handled the scenerio

Reply:
A better option is to have the encryption implement its own ChangeFeedObserver class (see https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/ChangeFeedObserver.java). You can also add a property into the ChangeFeedProcessorBuilderImpl to toggle between regular CFP and encrypted CFP when calling the "factory" class to create the respective instances.

Reply:
@milismsft Encryption have to access the ChangeFeedProcessorBuilderImpl which is implementation class in cosmos project, also ChangeFeedProcessorBuilderImpl will have to handle encryption logic, we want cosmos sdk to have minimal encryption logic .
IMA if we can avoid cosmos dependency on encryption project and treat encryption project as a wrapper as much as possible that would be good.
Coming back to this blocking call, I think ideal solution would have been the async consumer in core sdk , but that would breaking change now.
What concern we have with the current approach, and if we have very strong opinion against current design , i am open to other approach as well.

Reply:
Discussed offline with @milismsft  we are good here from this PR scope, however created work item to add support in change feed processor to support customer observer https://github.com/Azure/azure-sdk-for-java/issues/23738